[INIT-4490] - Schema Registry Private Link - AWS GA multi region (#536)
confluentinc/terraform-provider-confluent: docs/data-sources/confluent_schema_registry_cluster.md

- `kind` - (Required String) A kind of the Schema Registry cluster, for example, `Cluster`.
- `rest_endpoint` - (Required String) The HTTP endpoint of the Schema Registry cluster, for example, `https://psrc-00000.us-west-2.aws.confluent.cloud`.
- `private_rest_endpoint` - (Required String) The private HTTP endpoint of the Schema Registry cluster, for example, `https://lsrc.us-west-2.aws.private.confluent.cloud`.

> - `private_regional_rest_endpoint` - (Required Map) The private regional HTTP endpoint map of the Schema Registry cluster.

Reviewer: ```suggestion
- `private_regional_rest_endpoints` - (Required Map) The private regional HTTP endpoint map of the Schema Registry cluster, for example:
```
{
    "us-central-1": "https://lsrc-abc.us-central-1.aws.private.confluent.cloud",
    "us-west-2": "https://lsrc-abc.us-west-2.aws.private.confluent.cloud"
}
```
```

Reviewer: The reasoning is
```
private_regional_rest_endpoints['us-west-2']
```
looks a bit better than
```
private_regional_rest_endpoint['us-west-2']
```
Especially given that the private_rest_endpoint attribute is a string, whereas private_regional_rest_endpoints is a dictionary.

- `catalog_endpoint` - (Required String) The Catalog endpoint of the Schema Registry cluster, for example, `https://psrc-y1113.us-west-2.aws.confluent.cloud`.
- `display_name` - (Required String) The name of the Schema Registry cluster, for example, `Stream Governance Package`.
- `resource_name` - (Required String) The Confluent Resource Name of the Schema Registry cluster, for example, `crn://confluent.cloud/organization=1111aaaa-11aa-11aa-11aa-111111aaaaaa/environment=env-abc123/schema-registry=lsrc-abc123`.
- `cloud` - (Required String) The cloud service provider that the Schema Registry cluster belongs to, for example, `AWS`.

> -> **Note:** Use the `environment[0]` prefix for referencing these attributes, for example, `data.confluent_schema_registry_cluster.example_using_name.environment[0].id`.

Reviewer 2: It might be a good idea to explain to customers on how to use these 3 types of endpoints?

Reviewer: I like @channingdong's suggestion! Is there an existing Confluent doc or something that will make it easier to understand whether users should use private_rest_endpoint vs private_regional_rest_endpoints?

Reviewer 2: ```suggestion
-> **Note:** `private_regional_rest_endpoints` supersedes `private_rest_endpoint` to provide region specific private endpoint for Schema Registry cluster, as a result of recent Confluent Cloud API migration.

Please refer to the latest API [documentation](https://docs.confluent.io/cloud/current/api.html#tag/Clusters-(srcmv3)/operation/getSrcmV3Cluster) for details. 
```

Reviewer: I like this idea!

Can we drop the
```
, as a result of recent Confluent Cloud API migration.
```
part?

- `kind` - (Required String) A kind of the Schema Registry cluster, for example, `Cluster`.
- `rest_endpoint` - (Required String) The HTTP endpoint of the Schema Registry cluster, for example, `https://psrc-00000.us-west-2.aws.confluent.cloud`.
- `private_rest_endpoint` - (Required String) The private HTTP endpoint of the Schema Registry cluster, for example, `https://lsrc.us-west-2.aws.private.confluent.cloud`.

>   `private_regional_rest_endpoints` - (Required Map) The private regional HTTP endpoint map of the Schema Registry cluster, for example:
>   `{
>   "us-central-1": "https://lsrc-abc123.us-central-1.aws.private.confluent.cloud",
>   "us-west-2": "https://lsrc-abc123.us-west-2.aws.private.confluent.cloud"
>   }`

Reviewer: ```suggestion
  `private_regional_rest_endpoints` - (Required Map) The private regional HTTP endpoint map of the Schema Registry cluster. For example, to reference the endpoint corresponding to the us-central-1 region, use private_regional_rest_endpoints["us-central-1"].
```
